### PR TITLE
Docs: Clarify the status of the wp-block-styles theme support, and its intent

### PR DESCRIPTION
**Pull request by andrewserong**
## What?

Update the docs for the `wp-block-styles` theme support to clarify its intent to be used in classic themes, and that block themes should be adding styles to `theme.json` instead of depending on the opinionated styles of this theme support feature.

## Why?

Thanks to @simison for flagging that the docs here don't nudge anyone toward baking desired styles in to `theme.json` for block themes. While it might be easier for themes to use this theme support, they're not guaranteed to work well with block themes or to avoid styling conflicts with rules from global styles.

## How?

Update the docs hosted at: https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#default-block-styles

## Testing Instructions

Does this read okay, and is the information accurate? I've based that changes here on discussions from @carolinan and @tellthemachines, in particular on https://github.com/WordPress/gutenberg/pull/57841#issuecomment-1926633293, but it'd be good to double-check that I got the overall idea right, as I could very well be missing some context about this theme support!

### CODE REVIEW

**Comment by carolinan** [comment]
Can there be a new subheading above this section, because it can still read as if 
"Without these rules, the block would result in a broken layout containing no columns at all." .... applies to wp-block-styles.

**Comment by andrewserong** [comment]
Oh, great point @carolinan! I've added an `Opinionated block styles` heading. Happy to update any of the wording, of course!

File: docs/how-to-guides/themes/theme-support.md [hunk 1]
```diff
@@ -52,14 +52,18 @@ add_action( 'after_setup_theme', 'mytheme_setup_theme_supported_features' );
 
 Core blocks include default structural styles. These are loaded in both the editor and the front end by default. An example of these styles is the CSS that powers the columns block. Without these rules, the block would result in a broken layout containing no columns at all.
 
-The block editor allows themes to opt-in to slightly more opinionated styles for the front end. An example of these styles is the default color bar to the left of blockquotes. If you'd like to use these opinionated styles in your theme, add theme support for `wp-block-styles`:
+### Opinionated block styles
+
+The block editor allows themes to opt-in to slightly more opinionated styles for the front end. An example of these styles is the default color bar to the left of blockquotes. If you'd like to use these opinionated styles in a classic theme, add theme support for `wp-block-styles`:
```
Comment (ramonjd) [inline docs/how-to-guides/themes/theme-support.md]:
```suggestion
The block editor allows themes to opt in to slightly more opinionated styles for the front end. An example of these styles is the default color bar to the left of blockquotes. If you'd like to use these opinionated styles in a classic theme, add theme support for `wp-block-styles`:
```

Verb.

Sorry!

**Review by ramonjd — APPROVED** [review]
Nice

File: docs/how-to-guides/themes/theme-support.md (hunk 1 above)
Comment (andrewserong) [reply docs/how-to-guides/themes/theme-support.md]: 🦅 👀 thank you!

**Comment by andrewserong** [comment]
Thanks for the review @ramonjd! This PR isn't urgent, so I'll leave it overnight in case there's any other feedback, and then will merge it in tomorrow.

**Pull request merged**